typings/css-properties.d.ts

```diff
@@ -1933,6 +1933,14 @@ export interface CSSProperties
     CSSPropertiesPseudo,
     CSSPropertiesLossy {}
 
-export type CSSPropertiesRecursive = CSSProperties | CSSPropertiesArray
+export type CSSPropertiesRecursive =
+  | CSSProperties
+  | CSSPropertiesArray
+  | CSSFunction
+
+// TODO: This could be made generic. Issue PR if you're so inclined!
+export interface CSSFunction {
```

Comment:
While this will work well, it might be easier to add support in the `glamorous-component.ts` file which would also make it simpler to add the typing for the Props argument and help keep the logic of that function in one file.

```ts
// glamorous-component.d.ts

export interface CSSFunction<Props> {
    (props: Props): CSSPropertiesRecursive | CSSFunction<Props> 
}

export interface ExtraGlamorousProps<Props> {
  ...
  css?: CSSPropertiesRecursive | CSSFunction<Props>
  ...
}
...
export interface GlamorousComponent<ExternalProps, Props>
  extends React.ComponentClass<ExternalProps & ExtraGlamorousProps<ExternalProps>>,
    GlamorousComponentFunctions<ExternalProps, Props> {}
...
export type GlamorousComponentProps<ExternalProps> = JSX.IntrinsicAttributes &
  JSX.IntrinsicClassAttributes<
    React.Component<ExtraGlamorousProps<ExternalProps> & ExternalProps, React.ComponentState>
```